openmc/data/library.py

```diff
@@ -48,19 +49,30 @@ def register_file(self, filename):
 
         Parameters
         ----------
-        filename : str
+        filename : str or Path
             Path to the file to be registered.
+            If an ``xml`` file, treat as the depletion chain file without
+            materials.
 
         """
-        # Support pathlib
-        # TODO: Remove when support is Python 3.6+ only
-        filename = str(filename)
-
-        with h5py.File(filename, 'r') as h5file:
-            filetype = h5file.attrs['filetype'].decode()[5:]
-            materials = list(h5file)
+        if not isinstance(filename, pathlib.Path):
+            path = pathlib.Path(filename)
+        else:
+            path = filename
+
+        if path.suffix == '.xml':
+            filetype = 'depletion_chain'
+            materials = []
+        elif path.suffix == '.h5':
+            with h5py.File(path, 'r') as h5file:
+                filetype = h5file.attrs['filetype'].decode()[5:]
+                materials = list(h5file)
+        else:
+            raise ValueError(
+                "File type {} not supported by {}"
+                .format(path.name, self.__class__.__name__))
 
-        library = {'path': filename, 'type': filetype, 'materials': materials}
+        library = {'path': path.name, 'type': filetype, 'materials': materials}
```

Comment:
Using `path.name` looks like it loses information. If I take an existing `cross_sections.xml` that doesn't have a depletion chain in it, load it with `DataLibrary.from_xml`, register a chain file, and then re-export it, the exported XML file doesn't include the full path. You can probably use `str(path)` here instead.

Alternatively (this would require more work), the dictionary can just store a `Path` object under the 'path' key instead of a string.

Reply:
Interesting. I'll add some unit tests to cover this case in the next push